web/src/main/frontend/src/components/TakeTestQuestionComponent.js

```diff
@@ -10,7 +10,21 @@ import {inject, observer} from 'mobx-react';
 class TakeTestQuestionComponent extends React.Component {
   //TODO: review iscekic, These two lines look like they belong in a lifecycle method.
   answers = this.props.newTestForm.$('answers').add();
+
+  prepareMultipleChoiceQuestion() {
+    const checkedAnswers = [...this.props.testAnswers.testQuestionAnswerIds];
+    const allAnswerIds = [...this.props.question.answers].map(obj => obj.id);
```

Comment:
```suggestion
    const allAnswerIds = this.props.question.answers.map(obj => obj.id);
```

`map` returns a new array without touching the mapped one.